packages/rmf-codegen/scripts/install-vercel-java.js

```diff
@@ -0,0 +1,10 @@
+const shelljs = require('shelljs');
+const { javaIsInstalled, abortIfError } = require('./helpers');
+
+if (!javaIsInstalled() && shelljs.which('amazon-linux-extras')) {
+  console.info('Installing Java with amazon-linux-extras.');
```

Comment:
Let's check for the `VERCEL` env variable, so that this script only runs in a Vercel environment. See https://github.com/commercetools/commercetools-docs-kit/pull/838#discussion_r564347300

Reply:
Yes, this should not happen generally on an amazon linux.  I would consider checking for the `CI=true` env variable instead, it's also set by Vercel and seems to be a consensus among CI vendors (circleCI etc).  

you can then check  for the `VERCEL` variable instead of the amazon linux executable check that requires shelljs.  

--- 

I general, I'm not even really a fan of doing automatic installation on any environment directly from the package install.  Would a more conservative option where we add it to the site specific build scripts suffice?  Most CI systems have better built-in or "canonical" ways to provide java.   E.g. in a GitHub action I would use the `setup-java` action.

Reply:
> I general, I'm not even really a fan of doing automatic installation on any environment directly from the package install.

Me neither. I thought this requirement came from you.

Then I would suggest to drop this and configure CI to install Java.

Reply:
I do like that we fully "simulate" the required CI setup and typical filesystem layout here in the kit repository to have better testing of the real use case, so a lot of this here is not useless and should actually be done IMHO.  

Also, checking for java being installed and warning about it at install time is good.  But the actual install better belongs into  a top level `vercel-build` NPM script that would then be something like  `amazon-linux-extras install java-openjdk11 && yarn build`.  Vercel automatically picks an NPM script that is called "vercel-build" over the default "build" if such is present.